### ISSUE TITLE
Out Of Memory

### ISSUE THREAD

**Issue by YueSangShuai** [ErgastiAlex/MARS#13]
When I train on a large dataset, it starts training successfully at first, but in the middle of the batches the GPU memory suddenly grows explosively, causing an OOM error.

**Comment by YueSangShuai** [comment]
# output_mrtd = self.text_encoder.bert(mrtd_input_ids,
        #                                     attention_mask=text1.attention_mask,
        #                                     encoder_hidden_states=image_embeds,
        #                                     encoder_attention_mask=image_atts,
        #                                     return_dict=True,
        #                                     )

The OOM always happens at this point

**Comment by ErgastiAlex** [comment]
Hi, 
You may try to add torch.cuda.empty_cache() or implementing some gradient checkpointing technique.
